buildSrc/src/main/groovy/org/elasticsearch/gradle/EmptyDirTask.groovy

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.*
+import org.gradle.internal.nativeintegration.filesystem.Chmod
+import java.io.File
+import javax.inject.Inject
+
+/**
+ * Creates an empty directory.
+ */
+class EmptyDirTask extends DefaultTask {
+  @Input
+  def dir
```

Comment:
Can we not use def? It doesn't exist anywhere else in the existing buildSrc classes. If we need to specify something as Object, then ok, but we should also try to set types. For example, create() below should be void, and getChmod() doesn't need def at all.